pkg/serverless/invocationlifecycle/trace_test.go

```diff
@@ -455,3 +466,88 @@ func TestLanguageTag(t *testing.T) {
 		assert.Equal(t, tc.expectedTag, executionSpan.Meta["language"]) // expected tag from runtime
 	}
 }
+
+func TestCapturePayloadAsTags(t *testing.T) {
+	nestedMap := map[string]interface{}{
+		"key1": "value1",
+		"key2": map[string]interface{}{
+			"key3":    3,
+			"key4":    true,
+			"keylist": []interface{}{1, 2, 3, "four", 5.5, `{"keyInsideSlice":"val7","age":84}`},
+		},
+		"innerJSONString": `{"key5":"value5","age":42}`,
+		"innerJSONBytes":  []byte(`{"key6":"value6","age":21}`),
+	}
+	expectingResultMap := map[string]string{
+		"test.key1":                          "value1",
+		"test.key2.key3":                     "3",
+		"test.key2.key4":                     "true",
+		"test.key2.keylist.0":                "1",
+		"test.key2.keylist.1":                "2",
+		"test.key2.keylist.2":                "3",
+		"test.key2.keylist.3":                "four",
+		"test.key2.keylist.4":                "5.5",
+		"test.key2.keylist.5.keyInsideSlice": "val7",
+		"test.key2.keylist.5.age":            "84",
+		"test.innerJSONString.key5":          "value5",
+		"test.innerJSONString.age":           "42",
+		"test.innerJSONBytes.key6":           "value6",
+		"test.innerJSONBytes.age":            "21",
+	}
+	metaMap := make(map[string]string)
+	executionSpan := &pb.Span{
+		Meta: metaMap,
+	}
+	capturePayloadAsTags(nestedMap, executionSpan, "test", 0, 10)
+	assert.Equal(t, executionSpan.Meta, expectingResultMap)
+}
+
+func TestCapturePayloadAsTagsMaxDepth(t *testing.T) {
+	nestedMap := map[string]interface{}{
+		"key1": "value1",
+		"key2": map[string]interface{}{
+			"key3": map[string]interface{}{
+				"nestedKey": "nestedVal",
+			},
+			"key4": true,
+		},
+		"key5": "value5",
+	}
+	expectingResultMap := map[string]string{
+		"test.key1":      "value1",
+		"test.key2.key3": "{\"nestedKey\":\"nestedVal\"}",
+		"test.key2.key4": "true",
+		"test.key5":      "value5",
+	}
+	metaMap := make(map[string]string)
+	executionSpan := &pb.Span{
+		Meta: metaMap,
+	}
+	capturePayloadAsTags(nestedMap, executionSpan, "test", 0, 2)
+	assert.Equal(t, executionSpan.Meta, expectingResultMap)
+}
+
+func TestCapturePayloadAsTagsNilCases(t *testing.T) {
+	testMap := map[string]interface{}{
+		"key1": nil,
+		"key2": map[string]interface{}{
+			"key3": nil,
+			"key4": true,
+		},
+		"emptyMap":  map[string]interface{}{},
+		"emptyList": []interface{}{},
+	}
+	expectingResultMap := map[string]string{
+		"test.key1":      "",
+		"test.key2.key3": "",
+		"test.key2.key4": "true",
+		"test.emptyMap":  "{}",
+		"test.emptyList": "[]",
+	}
+	metaMap := make(map[string]string)
+	executionSpan := &pb.Span{
+		Meta: metaMap,
+	}
+	capturePayloadAsTags(testMap, executionSpan, "test", 0, 10)
+	assert.Equal(t, executionSpan.Meta, expectingResultMap)
+}
```

Comment:
Could we also add a specific test to make sure `{}` becomes `"function.response": "{}"`, `[]` becomes `"function.response": "[]"`, `nil` becomes `"function.response": null`, etc?